### PR TITLE
Remove the deprecated load_fractures_compilation function (deprecated since v0.6.0)

### PR DESCRIPTION
**Pull request by michaelgrund**
**Description of proposed changes**

Related to #2302

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: pygmt/datasets/samples.py [hunk 1]
```diff
@@ -25,7 +25,6 @@ def list_sample_data():
     names = {
         "bathymetry": "Table of ship bathymetric observations off Baja California",
         "earth_relief_holes": "Regional 20 arc-minutes Earth relief grid with holes",
-        "fractures": "Table of hypothetical fracture lengths and azimuths",
```
Comment (willschlitzer) [inline pygmt/datasets/samples.py:28]: I don't think this should be removed; we still want it in the names listed with `list_sample_data()`.

File: pygmt/datasets/samples.py [hunk 2]
```diff
@@ -272,14 +262,6 @@ def load_fractures_compilation(**kwargs):
...
-    if "suppress_warning" not in kwargs:
-        warnings.warn(
-            "This function has been deprecated since v0.6.0 and will be "
-            "removed in v0.9.0. Please use "
-            "load_sample_data(name='fractures') instead.",
-            category=FutureWarning,
-            stacklevel=2,
-        )
     fname = which("@fractures_06.txt", download="c")
     data = pd.read_csv(fname, header=None, sep=r"\s+", names=["azimuth", "length"])
```
Comment (willschlitzer) [inline pygmt/datasets/samples.py:266]:
```suggestion
    data = pd.read_csv(fname, header=None, delim_whitespace=True, names=["azimuth", "length"])
```
Keeping this in line with what @seisman suggested in #2301.

File: pygmt/datasets/samples.py [hunk 3]
```diff
@@ -250,20 +250,10 @@ def load_usgs_quakes(**kwargs):
...
+def _load_fractures_compilation():
     """
-    (Deprecated) Load a table of fracture lengths and azimuths as
-    hypothetically digitized from geological maps as a pandas.DataFrame.
-
-    .. warning:: Deprecated since v0.6.0. This function has been replaced with
-       ``load_sample_data(name="fractures")`` and will be removed in
-       v0.9.0.
-
-    This is the ``@fractures_06.txt`` dataset used in the GMT tutorials.
```
Comment (yvonnefroehlich) [inline pygmt/datasets/samples.py:262]: Maybe we should keep this background information of the data?

File: pygmt/datasets/samples.py [hunk 4]
```diff
@@ -250,20 +250,10 @@ def load_usgs_quakes(**kwargs):
...
-
-    The data are downloaded to a cache directory (usually ``~/.gmt/cache``) the
-    first time you invoke this function. Afterwards, it will load the data from
-    the cache. So you'll need an internet connection the first time around.
+    Load a table of fracture lengths and azimuths as hypothetically digitized
+    from geological maps as a pandas.DataFrame.
```
Comment (yvonnefroehlich) [inline pygmt/datasets/samples.py:256]:
```suggestion
    Load a table of fracture lengths and azimuths as hypothetically digitized
    from geological maps as a pandas.DataFrame.
    
    This is the "@fractures_06.txt" dataset used in the GMT tutorials.
```

File: pygmt/datasets/samples.py [hunk 5]
```diff
@@ -255,16 +245,10 @@ def load_fractures_compilation(**kwargs):
         columns.
```
Comment (seisman) [inline pygmt/datasets/samples.py:245]: Like the load_japan_quakes function, maybe we should say the returned column names are xxxx and xxxx.

Comment (michaelgrund) [reply pygmt/datasets/samples.py]: Done!

**Review by seisman — APPROVED** [review]
Looks good to me.

**Review by yvonnefroehlich — APPROVED** [review]

**Pull request merged**